### PR TITLE
assert that we don't forget started views

### PR DESCRIPTION
**Pull request by matklad**
## Pre-merge checklist

* [X] I am very sure this PR could not affect performance.

### CODE REVIEW

File: src/vsr/replica.zig [hunk 1]
```diff
@@ -2997,6 +2997,10 @@ pub fn ReplicaType(
         }
 
         fn ignore_prepare_ok(self: *Self, message: *const Message) bool {
+            if (self.primary_index(message.header.view) == self.replica) {
+                assert(message.header.view <= self.view);
```
Comment (jorangreef) [inline src/vsr/replica.zig:3001]: Could this fail if the replica is recovering after a crash (i.e. not in normal status, see below) and then receives a `prepare_ok`?

Comment (matklad) [reply src/vsr/replica.zig:3001]: That is exactly the case I am aiming to catch here. If I am recovering, and I receive a prepare_ok **for a view where I am the primary**, and I _don't_ know about the view, something is seriously wrong somewhere. 

This means that I started this "future" view and then forgot about it.

Comment (matklad) [reply src/vsr/replica.zig:3001]: (and we have "never forget about views" as our safety invariant after we removed recovery protocol)

Comment (jorangreef) [reply src/vsr/replica.zig:3001]: Yes, I mean, what if we are still recovering from the write ahead log?

Comment (jorangreef) [reply src/vsr/replica.zig:3001]: i.e. Some situation at startup where we do have the latest view on disk, but we're still getting there...

Comment (matklad) [reply src/vsr/replica.zig:3001]: To maybe rephrase this, the problem is not the `prepare_ok` per se. Rather, `prepare_ok` is an evidence that we as a primary have sent `start_view` for this view some time in the past. 

What I want to assert is "if a replica starts a view, and then restarts, it remembers that it did in fact start view". I don't think we can observe invariant this _directly_, but receiving a `prepare_ok` for future view is one case we _can_ assert. 


Practically, this wants to catch cases where duo to operator's error there are replica clones.

Comment (matklad) [reply src/vsr/replica.zig:3001]: Missed to replies!

>i.e. Some situation at startup where we do have the latest view on disk, but we're still getting there...

My understanding is that we immediately start with the latest view, there's no "getting there" in terms of the view number:

https://github.com/tigerbeetledb/tigerbeetle/blob/413de7bde379ddf67b6b2918113c8e90906847b3/src/vsr/replica.zig#L4877-L4880

https://github.com/tigerbeetledb/tigerbeetle/blob/413de7bde379ddf67b6b2918113c8e90906847b3/src/vsr/replica.zig#L562-L563

Comment (sentientwaffle) [reply src/vsr/replica.zig:3001]: > Yes, I mean, what if we are still recovering from the write ahead log?

The other thing to be aware of is that the replica does not process any messages (from the message bus) until it has opened its superblock (recovering its view number) and recovered its whole WAL.
